### PR TITLE
Fix failure in pytorch `pickle_module` test

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>

## What changes are proposed in this pull request?

Fix failure in pytorch `pickle_module` test.

## How is this patch tested?

Exisiting tests + fixed test

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/pytorch/test_pytorch_model_export.py [hunk 1]
```diff
@@ -817,35 +817,12 @@ def test_load_model_allows_user_to_override_pickle_module_via_keyword_argument(
...
-        return mlflow_torch_pickle_load(*args, **kwargs)
-
-    log_messages = []
-
-    def custom_warn(message_text, *args, **kwargs):
-        log_messages.append(message_text % args % kwargs)
-
-    with mock.patch(
-        "mlflow.pytorch.pickle_module.Unpickler"
-    ) as mlflow_torch_pickle_load_mock, mock.patch("mlflow.pytorch._logger.warning") as warn_mock:
```
Comment (harupy) [inline tests/pytorch/test_pytorch_model_export.py:835]: In this test, we check `Unpickler` is instantiated in `torch.load` but this approach no longer works in pytorch 1.9.0. See this PR for details.

https://github.com/pytorch/pytorch/pull/53139/

Comment (harupy) [reply tests/pytorch/test_pytorch_model_export.py:835]: As a workaround, we just check `torch.load` is called with the given `pickle_module`.

File: tests/pytorch/test_pytorch_model_export.py [hunk 2]
```diff
@@ -817,35 +817,12 @@ def test_load_model_allows_user_to_override_pickle_module_via_keyword_argument(
...
-    assert any(
-        [
-            "does not match the pickle module that was used to save the model" in log_message
-            and pickle.__name__ in log_message
-            and mlflow_pytorch_pickle_module.__name__ in log_message
-            for log_message in log_messages
-        ]
-    )
+        torch_load_mock.assert_called_with(mock.ANY, pickle_module=mlflow_pytorch_pickle_module)
+        warn_mock.assert_any_call(mock.ANY, mlflow_pytorch_pickle_module.__name__, pickle.__name__)
```
Comment (harupy) [inline tests/pytorch/test_pytorch_model_export.py:825]: Simplified a warninng message assertion using `assert_any_call`.

**Comment by harupy** [comment]
The `sagemaker` failed with a known issue so we can ignore it.

**Review by WeichenXu123 — APPROVED** [review]

**Pull request merged**